superx/information_extractors/item_info_extractor.py

```diff
@@ -0,0 +1,248 @@
+import requests
+from bs4 import BeautifulSoup
+from datetime import datetime
+import gzip
+import xml.etree.ElementTree as ET
+from superx.models import Product, BranchPrice
+from superx.app import db
+import logging
+from decimal import *
+
+logging.basicConfig(filename='info-extractor.log', level=logging.INFO,
+                    format='%(asctime)s: %(funcName)s: %(levelname)s: %(message)s')
+
+
+class InfoExtractor:
+    """
+    This class executes the following task:
+        1. web scrapes the url's from the url list and retrieves the relevant gzip file links
+        2. extracts the xml file from the gzip file and parses it into an xml tree
+        3. the relevant information is then extracted from the parsed xml tree
+        4. the information is placed in the relevant table in the db.
+            - this script only updates the product information (db table name = products)
+            - and branch price information (db table name = branch_price)
+
+        Currently this class can only handle the supermarkets: shufersal, mega and victory
+        XML files are found @https://www.consumers.org.il/item/transparency_price
+    """
+
+    def __init__(self):
+        self.current_super = ''
+        # list of unwanted names to be filter out
+        self.exclude_names = ['משלוחים', 'ריק', 'פיקדון', 'תיבה', 'משלוח']
+        self.super_specific_information = {'mega': {'store_name': 'mega',
```

Comment:
This dictionary is used in multiple places, can it go into a centralized place and imported into the modules that need it?

Reply:
I can make a yaml or json file with the dictionary, and  use that. is that what you mean ?

Reply:
I meant more like putting the dictionary in the __init__.py or app.py and importing it:
From app import specific_super_info

Reply:
ok ill talk with oded and aryeh

Reply:
I moved the dictionary to app.py, it is in the branch_extracter pull request.